695_Max_Area_of_Island.py

```diff
@@ -0,0 +1,33 @@
+# n = len(grid)
+# m = len(grid[0])
+# time: O(n*m)
+# space: O(1)
+from typing import List
+
+
+class Solution:
+    def maxAreaOfIsland(self, grid: List[List[int]]) -> int:
+        row, col = len(grid), len(grid[0])
+        max_area = 0
+
+        def check(row: int, col: int) -> int:
```

Comment:
The argument names of the check function are the same as the variables defined above. This is so confusing that you should use other words.
Also, the function name is unclear. Let's make it more descriptive.